### PR TITLE
fozzie-components@1.2.0 – Consoildate utility methods in f-header and f-footer

### PR DESCRIPTION
**Pull request by smilburn1990**
### Changed
- Service has been made to handle duplicate code in header and footer.
- Gitignore updated
- Vue for header and footer changed and methods removed

## TODO
- Write tests for services

👍

### CODE REVIEW

**Review by BanksySan — DISMISSED** [review]

**Review by dandel10n — CHANGES_REQUESTED** [review]
@smilburn1990 hey!

LGTM only few small things:

- I think that `@babel/plugin-proposal-optional-chaining` should be added as a dev dependency on mono repo root level package.json to avoid duplication in children packages.
- Maybe we need to make a folder for shared services? Now we have `packages/f-header`,  `packages/f-footer` etc and a js file with shared service.
- And a conflict needs to be resolved

File: services/shared.service.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+
```
Comment (ashleynolan) [inline services/shared.service.js]: Looks like there's a duplicate of this file here and at the base of `/packages`?

File: services/shared.service.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+/**
```
Comment (ashleynolan) [inline services/shared.service.js]: Also, would be good to have some tests for these service functions (might be worth putting in a TODO, unless you fancy making these as realise you're just moving them).

File: services/shared.service.js (hunk 1 above)
Comment (smilburn1990) [reply services/shared.service.js]: Still WIP @ashleynolan - sorry my bad. Will let you know when it needs review 👍

File: services/shared.service.js (hunk 2 above)
Comment (smilburn1990) [reply services/shared.service.js]: Can we get a ticket created? Who is the best person to speak to? @ashleynolan

Comment (ashleynolan) [reply services/shared.service.js]: We usually just create the tickets ourselves as we come across them – would add this to the [header epic](https://jira.just-eat.net/browse/WCP-107).  If you don't have access, let Darren know and he can make sure you have access.

Comment (smilburn1990) [reply services/shared.service.js]: @ashleynolan - i will add one now, can you approve this when you get a moment, would be nice to move some stuff over today :)

**Review by ashleynolan — DISMISSED** [review]
LGTM – think it just needs @dandel10n to re-review.

**Review by DamianMullins — DISMISSED** [review]
Needs a version bump and changelog entry.

**Comment by dandel10n** [comment]
@DamianMullins @ashleynolan  this change is related more to monorepo structure, moving repeated logic out of component level, than to one of the packages. Do you think it is worth to update both `f-footer` and `f-header` versions and changelogs?

**Comment by ashleynolan** [comment]
@dandel10n If it's more of a mono-repo change, then the CHANGELOG in the root should be updated and the package.json there updated. If there's changes to header/footer though too, they'll also need a CHANGELOG entry and package bump as it should be a versioned release for both of them.

We probably need to document the process for changes like this but in essence, all packages that change should have a `version` and `CHANGELOG` entry so that it's traceable (mainly for if a bug occurs and we need to look back at version history).

**Review by dandel10n — APPROVED** [review]

**Review by ashleynolan — APPROVED** [review]

File: .gitignore [hunk 3]
```diff
@@ -24,6 +24,7 @@ logs
 *.log
 .yalc
 yalc.lock
+package-lock.json
```
Comment (SachaZvetelman) [inline .gitignore:27]: We shouldn't be ignoring this file, in my opinion.

Comment (ashleynolan) [reply .gitignore:27]: We should have a `yarn.lock`, so it shouldn't be an issue to ignore the `package-lock.json`.

**Pull request merged**